### ISSUE TITLE
Passing bogus JSON triggers a warning/exception

### ISSUE THREAD

**Issue by pascaldevink** [overblog/GraphQLBundle#107]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Version/Branch   |  master#eed88d8e88f1fe7dea3835ab7371f146a6806543

When doing a valid JSON request, with a valid JSON body, a warning is thrown if the JSON body isn't valid GraphQL.

The message is `Warning: mb_strlen() expects parameter 1 to be string, array given`, in file `webonyx/graphql-php/src/Language/Source.php` line 24.
You can find the entire stack trace here: https://gist.github.com/pascaldevink/413fdd9d9436e138150ff14d6ae022ea

I guess there isn't enough error checking in `Overblog\GraphQLBundle\Request\Parser::getParsedBody()` when it come to JSON data. In fact, I'm not sure why it is supported, I can't find any documentation on how to send data this way.

Of course, it's a warning, which will be ignored in production, but I think it should be handled properly nonetheless.

**Comment by mcg-web** [comment]
Hi @pascaldevink can you give me a example of your request please?

**Comment by pascaldevink** [comment]
Yes, of course, stupid of me to forget:
```
curl -X "POST" "<host>" \
     -H "Content-Type: application/json" \
     -d $'{
  "query": {
    "foo": "bar"
  }
}'
```

**Comment by mcg-web** [comment]
ok the right way to write is:

```bash
curl -X "POST" "<host>" \
     -H "Content-Type: application/json" \
     -d $'{
  "query": "{foo: bar}",
}'
```
**query** value must be a string.
**variables** a json

I'll add a check to send a better error message and some documentation...

**Comment by mcg-web** [comment]
closed in favor of #252
